examples/examples/sloggish/sloggish_collector.rs

```diff
@@ -265,4 +266,8 @@ impl Collect for SloggishCollector {
         // TODO: GC unneeded spans.
         false
     }
+
+    fn current_span(&self) -> Current {
+        Current::unknown()
```

Comment:
in this example, the collector *does* know the current span's ID: https://github.com/tokio-rs/tracing/blob/f867e83dbaaa9ae73d7ee26a6766bb9b49730d06/examples/examples/sloggish/sloggish_collector.rs#L58

so, maybe this should return that span? this would require a slightly larger change, modifying [the `Span` struct](https://github.com/tokio-rs/tracing/blob/f867e83dbaaa9ae73d7ee26a6766bb9b49730d06/examples/examples/sloggish/sloggish_collector.rs#L66-L69) to also include a `&'static Metadata<'static>` that's stored in `new_span`, but then we could write something like this:
```suggestion
        if let Some(id) = self.current.id() {
            let metadata = self.spans
                .lock()
                .unwrap()
                .get(id)
                .unwrap_or_else(|| panic!("no metadata stored for span with ID {:?}", id))
                .metadata;
            return Current::new(id, metadata);
        }
        
        Current::none()
```